app/scripts/models/account.js

```diff
@@ -145,10 +145,30 @@ define([
       return this.get('verified') && ! this.get('accessToken');
     },
 
+    _generateAssertionPromise: null,
+    _generateAssertion: function () {
+      var self = this;
+
+      // share assertions between all concurrent assertion requesters. This
+      // eliminates multiple calls being made to sign certificates.
+      // See #3085
+      if (self._generateAssertionPromise) {
+        return self._generateAssertionPromise;
+      }
+
+      self._generateAssertionPromise =
```

Comment:
Seems like we'd want to handle if the sessionToken changes. Should we clear it? Cancel pending requests? Or store *n* promises indexed by sessionToken?